From: Prabhakar pujeri <Prabhakar.Pujeri@dell.com>
Date: Tue, 15 Oct 2019 14:24:46 +0200
Subject: dmidecode: Fix System Slot Information for PCIe SSD
Git-commit: fd08479625b5845e4d725ab628628f7ebfccc407
Patch-mainline: yes

Output for type 9 show <out of spec> for PCIe SSD. SMBIOS spec table
48 describes 2.5" and 3.5" PCIe SSD formats.

Signed-off-by: Prabhakar pujeri <prabhakar.pujeri@dell.com>
Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 dmidecode.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/dmidecode.c
+++ b/dmidecode.c
@@ -1900,10 +1900,12 @@ static const char *dmi_slot_length(u8 co
 		"Other", /* 0x01 */
 		"Unknown",
 		"Short",
-		"Long" /* 0x04 */
+		"Long",
+		"2.5\" drive form factor",
+		"3.5\" drive form factor" /* 0x06 */
 	};
 
-	if (code >= 0x01 && code <= 0x04)
+	if (code >= 0x01 && code <= 0x06)
 		return length[code - 0x01];
 	return out_of_spec;
 }
